src/test/scala/edu/ie3/simona/io/result/ResultEntityKafkaSpec.scala

```diff
@@ -0,0 +1,198 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+ */
+
+package edu.ie3.simona.io.result
+
+import akka.actor.ActorSystem
+import akka.testkit.TestActorRef
+import com.sksamuel.avro4s.RecordFormat
+import com.typesafe.config.ConfigFactory
+import edu.ie3.datamodel.models.result.NodeResult
+import edu.ie3.simona.event.ResultEvent.PowerFlowResultEvent
+import edu.ie3.simona.event.listener.ResultEventListener
+import edu.ie3.simona.io.result.plain.PlainResult.PlainNodeResult
+import edu.ie3.simona.io.result.plain.PlainWriter
+import edu.ie3.simona.test.KafkaSpecLike
+import edu.ie3.simona.test.KafkaSpecLike.Topic
+import edu.ie3.simona.test.common.{TestKitWithShutdown, UnitSpec}
+import edu.ie3.simona.util.ResultFileHierarchy
+import edu.ie3.util.quantities.PowerSystemUnits
+import edu.ie3.util.scala.io.ScalaReflectionSerde
+import io.confluent.kafka.serializers.AbstractKafkaSchemaSerDeConfig.SCHEMA_REGISTRY_URL_CONFIG
+import org.apache.kafka.clients.consumer.KafkaConsumer
+import org.apache.kafka.common.TopicPartition
+import org.apache.kafka.common.serialization.{Deserializer, Serdes}
+import org.apache.kafka.common.utils.Bytes
+import org.scalatest.GivenWhenThen
+import org.scalatest.concurrent.Eventually
+import tech.units.indriya.quantity.Quantities
+
+import java.time.ZonedDateTime
+import java.util.UUID
+import scala.concurrent.duration._
+import scala.jdk.CollectionConverters._
+import scala.jdk.DurationConverters._
+import scala.language.postfixOps
+
+/** Adapted from
+  * https://kafka-tutorials.confluent.io/produce-consume-lang/scala.html
+  */
+class ResultEntityKafkaSpec
+    extends TestKitWithShutdown(
+      ActorSystem(
+        "ResultEntityKafkaSpec",
+        ConfigFactory
+          .parseString(
+            """akka.loggers = ["edu.ie3.simona.test.common.SilentTestEventListener"]
+          |akka.loglevel="info"
+          """.stripMargin
+          )
+      )
+    )
+    with UnitSpec
```

Comment:
unit spec is also in TestKitWithShutdown so might be removed here

Reply:
Good catch